src/global/global_cuda.cu

```diff
@@ -9,16 +9,12 @@
 bool memory_allocated;
 Real *dev_conserved, *dev_conserved_half;
 Real *Q_Lx, *Q_Rx, *Q_Ly, *Q_Ry, *Q_Lz, *Q_Rz, *F_x, *F_y, *F_z;
+Real *ctElectricFields;
 Real *eta_x, *eta_y, *eta_z, *etah_x, *etah_y, *etah_z;
-Real *dev_dti;
```

Comment:
Is the time step reduction fixed so that we don't need this any more?

Reply:
The time step reduction is fixed. I replace the atomic with the built in one on HIP and some fancy bit casting shenanigans on CUDA. The CUDA atomic is taken from CUML.